solution-base/build.sh

```diff
@@ -0,0 +1,182 @@
+#!/bin/bash
+set -eu
+
+PWD=$(pwd)
+BUILD_ROOT=${PWD}/_build
+ISO_ROOT=${BUILD_ROOT}/root
+IMAGES_ROOT=${ISO_ROOT}/images
+
+PRODUCT_NAME=Zenko-Base
+PRODUCT_LOWERNAME=zenko-base
+BUILD_TIMESTAMP=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+BUILD_HOST=$(hostname)
+
+VERSION_SHORT=$(git describe --abbrev=0)
+VERSION_FULL=${VERSION_SHORT}-dev
+GIT_REVISION=$(git describe --long --always --tags --dirty)
+ISO=${BUILD_ROOT}/${PRODUCT_LOWERNAME}-${VERSION_FULL}.iso
+
+DOCKER=docker
+DOCKER_OPTS=
+DOCKER_SOCKET=${DOCKER_SOCKET:-unix:///var/run/docker.sock}
+HARDLINK=hardlink
+SKOPEO=skopeo
+SKOPEO_OPTS="--override-os linux --insecure-policy"
+OPERATOR_TAG=$(grep /operator: deps.txt | awk -F ':' '{print $2}')
+SOLUTION_REGISTRY=metalk8s-registry-from-config.invalid/${PRODUCT_LOWERNAME}-${VERSION_FULL}
+
+KUBEDB_SCRIPT_BRANCH_TAG=89fab34cf2f5d9e0bcc3c2d5b0f0599f94ff0dca
+
+export KUBEDB_OPERATOR_TAG=${OPERATOR_TAG}
+export KUBEDB_SCRIPT_LOCATION="curl -fsSL https://raw.githubusercontent.com/kubedb/installer/${KUBEDB_SCRIPT_BRANCH_TAG}/"
+export KUBEDB_NAMESPACE=SOLUTION_ENV
+export KUBEDB_SERVICE_ACCOUNT=kubedb-operator
+export KUBEDB_OPERATOR_NAME=operator
+export KUBEDB_ENABLE_RBAC=true
+export KUBEDB_RUN_ON_MASTER=0
+export KUBEDB_ENABLE_VALIDATING_WEBHOOK=false
+export KUBEDB_ENABLE_MUTATING_WEBHOOK=false
+export KUBEDB_DOCKER_REGISTRY=${SOLUTION_REGISTRY}
+export KUBEDB_IMAGE_PULL_POLICY=IfNotPresent
+export KUBEDB_ENABLE_ANALYTICS=false
+export KUBEDB_ENABLE_STATUS_SUBRESOURCE=false
+export KUBEDB_BYPASS_VALIDATING_WEBHOOK_XRAY=false
+export KUBEDB_USE_KUBEAPISERVER_FQDN_FOR_AKS=true
+export KUBEDB_PRIORITY_CLASS=system-cluster-critical
+
+# grab our dependencies from our deps.txt file as an array
+readarray -t DEP_IMAGES < deps.txt
+
+function clean()
+{
+    echo cleaning
+    rm -rf ${BUILD_ROOT}
+    rm -rf ca.crt ca.key server.crt server.key
+}
+
+function mkdirs()
+{
+    echo making dirs
+    mkdir -p ${ISO_ROOT}
+    mkdir -p ${IMAGES_ROOT}
+}
+
+function kubedb_yamls()
+{
+    echo downloading kubedb yamls
+    OPERATOR_PATH=${BUILD_ROOT}/operator.yaml
+
+    yamls=(
+        service-account
+        rbac-list
+        user-roles
+        appcatalog-user-roles
+        psp/operator
+        psp/mongodb
+        psp/redis
+        kubedb-catalog/mongodb
+        kubedb-catalog/redis
```

Comment:
I think the catalog yaml files should be either provided by us directly or patched because they're what define the images - and the offline nature of the iso makes it necessary to point the local registry for these images.

Reply:
the yaml files included in kubedbs install process include variable substitution that I modify using the KUBEDB_* values above, the image name is using the metalk8s prefixed image paths. I also specify a SOLUTION_ENV that will need to be replaced during the install process ( will probably want a readme for this later on ).

Reply:
Ah sorry, I missed that.